scripts/migrate_project_mailing_lists.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+"""Create mailing lists for all top-level projects
+"""
+import logging
+import sys
+
+from modularodm import Q
+
+from framework.transactions.context import TokuTransaction
+
+from website.app import init_app
+from website.mailing_list.utils import create_list as real_create_list
+from website.models import Node
+from website.notifications.model import NotificationSubscription
+from website.notifications.utils import to_subscription_key
+from scripts import utils as script_utils
+
+logger = logging.getLogger(__name__)
+
+EVENT = 'mailing_list_events'
+SUBSCRIPTION_TYPE = 'email_transactional'
+
+def fake(*args, **kwargs):
+    # Override actual method to prevent outgoing calls
+    return
+
+def get_targets():
+    return Node.find()
+
+def migrate(dry_run=True):
+    if dry_run:
+        create_list = fake
+    else:
+        create_list = real_create_list
+
+    successful_enables = []
+    successful_disables = []
+    unknown_failures = {}
+    nodes = get_targets()
+    ncount = len(nodes)
+    logger.info('Preparing to migrate {} nodes.'.format(ncount))
+    for i, node in enumerate(nodes):
+        if node.is_registration or node.is_collection or node.is_deleted:
+            try:
+                logger.info('({0}/{1})Disabling mailing list for registration/dashboard {2}'.format(i+1, ncount, node._id))
+                node.mailing_enabled = False    
+                node.save()
+                successful_disables.append(node._id)
+            except Exception as e:
+                unknown_failures[node._id] = e
+        else:
+            try:
+                subscription = NotificationSubscription(
+                    _id=to_subscription_key(node._id, EVENT),
+                    owner=node,
+                    event_name=EVENT
+                )
+                logger.info('({0}/{1})Enabling mailing list for node {2}'.format(i+1, ncount, node._id))
+                node.mailing_enabled = True
+
+                for user in node.contributors:
+                    if user.is_active:
+                        logger.info('Subscribing user {} on node {}'.format(user, node))
+                        subscription.add_user_to_subscription(user, SUBSCRIPTION_TYPE)
+                        # users added on `create_list`
+
+                subscription.save()
+                node.save()
+                successful_enables.append(node._id)
+            except Exception as e:
+                unknown_failures[node._id] = e
+            else:
+                create_list(node._id)
+        if i % 100 == 0:
```

Comment:
Clears the cache so that it doesn't crash migrating the ~70K nodes on prod. Something similar was done when the Node/Nodelog backref removal script was run.